runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowRunner.java

```diff
@@ -1451,24 +1455,19 @@ void recordViewUsesNonDeterministicKeyCoder(PTransform<?, ?> ptransform) {
     }
   }
 
-  private interface PubsubUnboundedSourceOverrider {
-    PubsubUnboundedSource getOverriddenTransform();
-  }
-
   /**
    * Suppress application of {@link PubsubUnboundedSource#expand} in streaming mode so that we can
    * instead defer to Windmill's implementation.
    */
-  private static class StreamingPubsubIORead extends PTransform<PBegin, PCollection<PubsubMessage>>
-      implements PubsubUnboundedSourceOverrider {
+  private static class StreamingPubsubIORead
+      extends PTransform<PBegin, PCollection<PubsubMessage>> {
```

Comment:
With portable submission, it would be preferable to not have any override, and handle it in the service. Mostly the overrides seem trivial so we don't need them in the SDK.